### PR TITLE
Feat/remove copy sender

### PR DESCRIPTION
**Pull request by guptapratykshh**
## Proposed Changes

 - Add sender/receiver support for hpx::remove_copy and hpx::remove_copy_if with scheduler-backed execution policies.
 - Preserve synchronous iterator returns for non-task scheduler policies and sender returns for task policies.
 - Add tests covering returned iterators, empty/all/none matching ranges, iterator categories, hpx::exception_list, and direct std::bad_alloc propagation.

## Checklist

Not all points below apply to all pull requests.

- [ ] I have added a new feature and have added tests to go along with it.
- [ ] I have fixed a bug and have added a regression test.
- [ ] I have added a test using random numbers; I have made sure it uses a seed, and that random numbers generated are valid inputs for the tests.

### CODE REVIEW

File: libs/core/algorithms/include/hpx/parallel/algorithms/remove_copy.hpp [hunk 1]
```diff
@@ -456,12 +456,41 @@ namespace hpx {
...
+                                    sequential_remove_copy_if(first, last, dest,
+                                        HPX_MOVE(pred), hpx::identity_v);
+                            }
+                            catch (...)
+                            {
+                                hpx::parallel::util::detail::
+                                    handle_local_exceptions<ExPolicy>::call(
+                                        std::current_exception());
+                            }
+                        })));
```
[the remark this answers is not in the document]
Comment (guptapratykshh) [reply libs/core/algorithms/include/hpx/parallel/algorithms/remove_copy.hpp]: This path cannot fall through: the selected one-argument handle_local_exceptions<ExPolicy>::call overload is [[noreturn]], including the unsequenced specialization, so adding throw; would be unreachable and redundant

Comment (hkaiser) [inline libs/core/algorithms/include/hpx/parallel/algorithms/remove_copy.hpp]: This implementation loses parallelism when run with a s&r executor. Is that intended?
